### DISCUSSION TITLE
Subscription Event Emails

### DISCUSSION THREAD

**Discussion by CustomCodeIT** [craftcms/commerce-stripe#264] [Ideas]
Hi there,

I have been through the documentation and can't see anything (unless I missed it), but it would be great to have some functionality around sending emails for subscription events (new subscription, canceled, plan change etc.)

I have a few clients that require custom transactional emails to be sent to customers. This ranges from welcome emails, onboarding  or user/plan specific instructions, plan change confirmation emails etc.

There's probably a bunch of ways to achieve this functionality on my own (which I am investigating) but it would be great to see this as part of the out-of-the-box experience. This could be similar to the order status email functionality that already exists in Commerce (but not specific to Stripe).

Some of the Stripe generated emails are ok, but we would like full control of the branding and content experience.

Maybe I've missed a really obvious solution here so happy to discuss!

**Comment by Saboteur777** [comment]
Hey @CustomCodeIT ! 

I do this on my SaaS with the Craft Notifications plugin: https://github.com/craftpulse/craft-notifications
Ideally: 
1. You create the content in a Twig file, 
2. Create the notification and configure it with a PHP file, 
3. Create a job to send the notification (optional, helps preventing occasional notification templating errors bubbling out to visitors)
4. The actual notification is sent, e.g.:
```
$template = new LimitReachedNotification($originId);
Notifications::getInstance()->notificationsService->send($template);
```

You can then listen to events (Stripe webhooks, user creation, etc.) and send the notification as above.
